node/network/availability-recovery/src/lib.rs

```diff
@@ -972,7 +1007,9 @@ impl AvailabilityRecoverySubsystem {
 				}
 				output = state.ongoing_recoveries.select_next_some() => {
 					if let Some((candidate_hash, result)) = output {
-						state.availability_lru.put(candidate_hash, result);
+						if let Ok(result) = CachedResult::try_from(result) {
```

Comment:
on the similar note, this will become

`if let Ok(validity) = CachedValidity::try_from(result) {`